src/Main/DeathRecapTracker.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import Abilities from 'Parser/Core/Modules/Abilities';
+import DEFENSIVE_BUFFS from 'common/DEFENSIVE_BUFFS';
+import SpellUsable from 'Parser/Core/Modules/SpellUsable';
+import Tab from 'Main/Tab';
+import DeathRecap from './DeathRecap';
+
+class DeathRecapTracker extends Analyzer {
+
+  deaths = [];
+  events = [];
+  healed = [];
+  damaged = [];
+  cooldowns = [];
+  buffs = [];
+  lastBuffs = [];
+
+  static dependencies = {
+    combatants: Combatants,
+    abilities: Abilities,
+    spellUsable: SpellUsable,
+  };
+
+  on_initialized() {
+    this.cooldowns = this.abilities.abilities.filter(e => 
+      (e.category === Abilities.SPELL_CATEGORIES.DEFENSIVE || e.category === Abilities.SPELL_CATEGORIES.SEMI_DEFENSIVE) &&
+      e.enabled === true
+    );
+    //add additional defensive buffs/debuffs to common/DEFENSIVE_BUFFS
+    this.buffs = DEFENSIVE_BUFFS.concat(this.cooldowns);
+  }
+
+  addEvent(event) {
+    const extendedEvent = event;
```

Comment:
This would still mutate the event being passed in, please use:
```js
const extendedEvent = { ...event };
```
or another similar shallow/deep clone method.